forms/core/CampaignMonitor.php

```diff
@@ -37,29 +37,41 @@ public function get_last_error(){
      */
     public function refresh_token($auth = array())
     {
-	   return $this->api->refreshToken();
+        $refreshToken = empty($auth) ? '' : $auth['refresh_token'];
+        return json_decode($this->api->refreshToken($refreshToken));
     }
     /**
      * @param array $auth override the class authentication credentials
      * @return mixed|null list of clients
      */
     public function get_clients($credentials = array())
     {
+	    if (empty($credentials)) {

```

Comment:
Is this intended? Have we checked the use of get_clients throughout the code that might not pass in the $credentials, for example update() within Application.php?

Reply:
This might implied that the code is relying on other method being called first, which is not very good design... its definitely a potential improvement.

Reply:
With the previous code, if you didn't have any credentials it wouldn't set anything and just call the API straight up, which would fail every time because you're not passing in the tokens or anything. I'm not sure why they do that though